File: tests/fixtures/cpp/sample.cpp
#include <iostream>
#include <vector>
#include <string>

namespace myapp {
    namespace core {
        /// <summary>
        /// Person class
        /// </summary>
        class Person {
        private:
            std::string name_;
            int age_;

        public:
            Person(const std::string& name, int age)
                : name_(name), age_(age) {}

            std::string GetName() const { return name_; }
            int GetAge() const { return age_; }

            virtual std::string ToString() const {
                return name_ + " (" + std::to_string(age_) + ")";
            }
        };

        template <typename T>
        class Repository {
        public:
            virtual T GetById(int id) = 0;
            virtual void Save(const T& entity) = 0;
            virtual ~Repository() = default;
        };

        enum class Status {
            Active,
            Inactive,
            Pending
        };

        struct Point {
            int x;
            int y;
        };

        std::unique_ptr<Person> CreatePerson(const std::string& name, int age);
    }
}

File: tests/fixtures/rs/sample.rs
/// Person module.
pub mod person {
    /// Person struct.
    pub struct Person {
        pub name: String,
        pub age: u32,
    }

    impl Person {
        pub fn new(name: String, age: u32) -> Self {
            Self { name, age }
        }

        pub fn greet(&self) -> String {
            format!("Hello, {}", self.name)
        }
    }
}

/// Repository trait.
pub trait Repository<T> {
    fn get_by_id(&self, id: u32) -> Result<T, String>;
    fn save(&self, entity: T) -> Result<(), String>;
}

/// Status enum.
pub enum Status {
    Active,
    Inactive,
    Pending,
}

/// Utility function.
pub fn create_person(name: &str, age: u32) -> person::Person {
    person::Person::new(name.to_string(), age)
}

/// Generic struct.
pub struct Container<T> {
    pub value: T,
}
